### PR TITLE
Support for a 'sequence_rendering' method 

### PR DESCRIPTION
**Pull request by geekscruff**
I'm working on [Hyku 1372](https://github.com/samvera-labs/hyku/issues/1372) and wanted a way of adding a 'rendering' element into the sequences.

I came up with `#manifest_extras` as a default place to extend iiif_manifest with extra elements. `#manifest_extras` would return a hash, which can be empty. 

For the current implementation, if the `#manifest_extras` hash contains `:sequence_rendering [array of rendering hashes]` these would be picked up by the sequence_builder and added into the manifest.

I thought this could also be extended easily, eg. `:metadata` could be provided to add metadata at manifest level.

### CODE REVIEW

**Comment by jcoyne** [comment]
@geekscruff Is it possible that `manifest_extras` could be optional?  Could you test for `respond_to?(:manifest_extras)` prior to calling it or provide a default implementation that can be overridden?

I'd also like to see @tpendragon @escowles @jpstroop @mejackreed or @cbeer have a look at this too.

Thanks for your work on this @geekscruff!

**Comment by geekscruff** [comment]
thanks @jcoyne ... adding `respond_to?` sounds good to me _ I was a little concerned about breaking the gem for others, and adding complexity some folks wouldn't need. I'll hang on for further feedback before making changes.

File: lib/iiif_manifest/manifest_builder/sequence_builder.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def sequence
             begin
               sequence = IIIF::Presentation::Sequence.new
               sequence["@id"] ||= work.manifest_url + "/sequence/normal"
+              sequence["rendering"] = work.manifest_extras[:sequence_rendering].each { | x | x.to_h } if work.respond_to?(:manifest_extras) && work.manifest_extras[:sequence_rendering].present?
```
Comment (jcoyne) [inline lib/iiif_manifest/manifest_builder/sequence_builder.rb]: would you mind putting this logic in a new private method?

File: spec/lib/iiif_manifest/manifest_factory_spec.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 require 'spec_helper'
 
 RSpec.describe IIIFManifest::ManifestFactory do
-  let(:presenter_class) { Book }
-  subject { described_class.new(book_presenter) }
-  let(:book_presenter) { presenter_class.new('book-77') }
+  let(:presenter_class) {Book}
```
Comment (jcoyne) [inline spec/lib/iiif_manifest/manifest_factory_spec.rb]: Please revert these spacing changes as they're part of the ruby style guide.

Comment (jcoyne) [inline spec/lib/iiif_manifest/manifest_factory_spec.rb]: If you rebase on master and run `rubocop -a` it should fix these automatically.

**Comment by tpendragon** [comment]
Sorry, I missed this. I don't feel great about `manifest_extras` - it feels like a catch-all solution that could quickly get nasty. I think I'd prefer if for the individual use cases for additional items to add to the manifest that we figure out the right interface. The use case here seems to be `rendering` - can we just add a `rendering` method as a `MAY`?

**Comment by geekscruff** [comment]
@tpendragon Now I've made it MAY, I prefer this solution. I'll make the changes.

**Comment by tpendragon** [comment]
@geekscruff Thanks!

**Comment by geekscruff** [comment]
@jcoyne I've made those changes, plus changed from 'manifest extras' to 'sequence rendering' method.

**Comment by geekscruff** [comment]
Gah, now I've broken the build. Will have to fix tomorrow.

**Comment by geekscruff** [comment]
@jcoyne @tpendragon is this looking OK now? I went with '#sequence_rendering' to couple the method to where it's added, 2nd commit should be just the merge from master, 1st is my changes. Thanks!

**Comment by tpendragon** [comment]
I'm cool with it. @jcoyne ?

**Pull request merged**